workmail-message-flow-state-machine/src/app.py

```diff
@@ -0,0 +1,168 @@
+            },
+            'recipients': list of strings,                          # Optional. Indicates list of recipients for which this action applies
+            'default': boolean                                      # Optional. Indicates whether this action applies to all recipients
+          }
+        ]}
+
+    """
+    logger.info(email_summary)
+    client = boto3.client('stepfunctions')
+    start_execution_response = client.start_execution(
+        stateMachineArn=state_machine_arn,
+        input=json.dumps(email_summary)
+    )
+
+    logger.info(start_execution_response)
+
+    time.sleep(1) # TODO retry/wait logic should be configurable or determined from average/maximum history of execution run time, also it should retry if execution is not complete based on get_execution_history
+    retry_count = 0
+    while 1:
+        
+        state_machine_execution_history = client.get_execution_history(
+            executionArn=start_execution_response['executionArn']
+        )
+        logger.info(state_machine_execution_history['events'])
+        
+        for state_machine_event in state_machine_execution_history['events']:
+            
+            if machine_state_for_output:
+                
+                if 'stateExitedEventDetails' in state_machine_event:
+                    this_state_name = state_machine_event['stateExitedEventDetails']['name']
+                    this_state_output = state_machine_event['stateExitedEventDetails']['output']
+                    if this_state_name == machine_state_for_output:
+                        if is_valid_workmail_output(this_state_output):
+                            return json.loads(this_state_output)
+                        else:
+                            # TODO
+                            logger.info("State output from {this_state_name} is not a valid response for WorkMail")
+                
+            elif state_machine_event['type'] == 'ExecutionSucceeded':
+                
+                if 'executionSucceededEventDetails' in state_machine_event:
+                    this_execution_output = state_machine_event['executionSucceededEventDetails']['output']
+                    if is_valid_workmail_output(this_execution_output):
+                        return json.loads(this_execution_output)
+                    else:
+                        logger.info("Execution output from the Step Function state machine is not a valid response for WorkMail")
+        
+        if retry_count > 10:
+            break
+        time.sleep(10)
+        
+    logger.info("Default action. Unable to retrieve output from Step Function state machine state or execution.")
+    return {
+        'actions': [
+            {
+                'allRecipients': True,                  # For all recipients
+                'action' : { 'type' : 'DEFAULT' }       # let the email be sent normally
+            }
+        ]
+    }
+        
+def is_valid_workmail_output(output):
```

Comment:
WorkMail already validates the value returned from the lambda function. If the returned value isn't in a valid format then WorkMail uses the `default action` on the email, this `default action` is specified by the user while setting up a synchronous email flow rule.